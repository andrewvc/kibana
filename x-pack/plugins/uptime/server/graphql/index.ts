/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { createMonitorsResolvers, monitorsSchema } from './monitors';
import { createPingsResolvers, pingsSchema } from './pings';
import { CreateUMGraphQLResolvers } from './types';
import { unsignedIntegerResolverFunctions, unsignedIntegerSchema } from './unsigned_int_scalar';
import { sortDirectionEnumSchema, sortDirectionResolvers } from './sort_direction_enum'

export { DEFAULT_GRAPHQL_PATH } from './constants';
export const resolvers: CreateUMGraphQLResolvers[] = [
  createPingsResolvers,
  unsignedIntegerResolverFunctions,
  createMonitorsResolvers,
  sortDirectionResolvers,
];
export const typeDefs: any[] = [pingsSchema, unsignedIntegerSchema, monitorsSchema, sortDirectionEnumSchema];
