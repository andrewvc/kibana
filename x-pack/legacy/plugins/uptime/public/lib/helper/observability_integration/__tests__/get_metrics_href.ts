/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import {
  getMetricsContainerHref,
  getMetricsKubernetesHref,
  getMetricsIpHref,
} from '../get_metrics_href';
import { MonitorSummary } from '../../../../../common/graphql/types';

describe('getMetricsHref', () => {
  let summary: MonitorSummary;
  beforeEach(() => {
    summary = {
      monitor_id: 'foo',
      state: {
        summary: {},
        checks: [
          {
            monitor: {
              ip: '151.101.202.217',
              status: 'up',
            },
            container: {
              id: 'test-container-id',
            },
            kubernetes: {
              pod: {
                uid: 'test-pod-uid',
              },
            },
            timestamp: '123',
          },
        ],
        timestamp: '123',
      },
    };
  });

  it('getMetricsContainerHref creates a link for valid parameters', () => {
    const result = getMetricsContainerHref(summary, 'foo');
    expect(result).toMatchSnapshot();
  });

  it('getMetricsContainerHref does not specify a base path when none is available', () => {
    expect(getMetricsContainerHref(summary, '')).toMatchSnapshot();
  });

  it('getMetricsContainerHref returns undefined when no container id is present', () => {
    summary.state.checks = [];
    expect(getMetricsContainerHref(summary, 'foo')).toBeUndefined();
  });

  it('getMetricsContainerHref returns the first item when multiple container ids are supplied', () => {
    summary.state.checks = [
      {
        monitor: {
          ip: '151.101.202.217',
          status: 'up',
        },
        container: {
          id: 'test-container-id',
        },
        kubernetes: {
          pod: {
            uid: 'test-pod-uid',
          },
        },
        timestamp: '123',
      },
      {
        monitor: {
          ip: '151.101.202.27',
          status: 'up',
        },
        container: {
          id: 'test-container-id-foo',
        },
        kubernetes: {
          pod: {
            uid: 'test-pod-uid-bar',
          },
        },
        timestamp: '123',
      },
    ];
    expect(getMetricsContainerHref(summary, 'bar')).toMatchSnapshot();
  });

  it('getMetricsContainerHref returns undefined when checks are undefined', () => {
    delete summary.state.checks;
    expect(getMetricsContainerHref(summary, '')).toBeUndefined();
  });

  it('getMetricsKubernetesHref creates a link for valid parameters', () => {
    const result = getMetricsKubernetesHref(summary, 'foo');
    expect(result).not.toBeUndefined();
    expect(result).toMatchSnapshot();
  });

  it('getMetricsKubernetesHref does not specify a base path when none is available', () => {
    expect(getMetricsKubernetesHref(summary, '')).toMatchSnapshot();
  });

  it('getMetricsKubernetesHref returns undefined when no pod data is present', () => {
    summary.state.checks = [];
    expect(getMetricsKubernetesHref(summary, 'foo')).toBeUndefined();
  });

  it('getMetricsKubernetesHref selects the first pod uid when there are multiple', () => {
    summary.state.checks = [
      {
        monitor: {
          ip: '151.101.202.217',
          status: 'up',
        },
        container: {
          id: 'test-container-id',
        },
        kubernetes: {
          pod: {
            uid: 'test-pod-uid',
          },
        },
        timestamp: '123',
      },
      {
        monitor: {
          ip: '151.101.202.27',
          status: 'up',
        },
        container: {
          id: 'test-container-id-foo',
        },
        kubernetes: {
          pod: {
            uid: 'test-pod-uid-bar',
          },
        },
        timestamp: '123',
      },
    ];
    expect(getMetricsKubernetesHref(summary, '')).toMatchSnapshot();
  });

  it('getMetricsKubernetesHref returns undefined when checks are undefined', () => {
    delete summary.state.checks;
    expect(getMetricsKubernetesHref(summary, '')).toBeUndefined();
  });

  it('getMetricsKubernetesHref returns undefined when checks are null', () => {
    summary.state.checks![0]!.kubernetes!.pod!.uid = null;
    expect(getMetricsKubernetesHref(summary, '')).toBeUndefined();
  });

  it('getMetricsIpHref creates a link for valid parameters', () => {
    const result = getMetricsIpHref(summary, 'bar');
    expect(result).toMatchSnapshot();
  });

  it('getMetricsIpHref does not specify a base path when none is available', () => {
    expect(getMetricsIpHref(summary, '')).toMatchSnapshot();
  });

  it('getMetricsIpHref returns undefined when ip is undefined', () => {
    summary.state.checks = [];
    expect(getMetricsIpHref(summary, 'foo')).toBeUndefined();
  });

  it('getMetricsIpHref returns undefined when ip is null', () => {
    summary.state.checks![0].monitor.ip = null;
    expect(getMetricsIpHref(summary, 'foo')).toBeUndefined();
  });

  it('getMetricsIpHref returns a url for ors between multiple ips', () => {
    summary.state.checks = [
      {
        timestamp: '123',
        monitor: {
          ip: '152.151.23.192',
          status: 'up',
        },
      },
      {
        monitor: {
          ip: '151.101.202.217',
          status: 'up',
        },
        container: {
          id: 'test-container-id',
        },
        kubernetes: {
          pod: {
            uid: 'test-pod-uid',
          },
        },
        timestamp: '123',
      },
    ];
    expect(getMetricsIpHref(summary, 'foo')).toMatchSnapshot();
  });

  it('getMetricsIpHref returns undefined if checks are undefined', () => {
    delete summary.state.checks;
    expect(getMetricsIpHref(summary, 'foo')).toBeUndefined();
  });
});
