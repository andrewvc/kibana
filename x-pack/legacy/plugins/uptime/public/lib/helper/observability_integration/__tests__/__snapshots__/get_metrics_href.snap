// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getMetricsHref getMetricsContainerHref creates a link for valid parameters 1`] = `"foo/app/infra#/link-to/container-detail/test-container-id"`;

exports[`getMetricsHref getMetricsContainerHref does not specify a base path when none is available 1`] = `"/app/infra#/link-to/container-detail/test-container-id"`;

exports[`getMetricsHref getMetricsContainerHref returns the first item when multiple container ids are supplied 1`] = `"bar/app/infra#/link-to/container-detail/test-container-id"`;

exports[`getMetricsHref getMetricsIpHref creates a link for valid parameters 1`] = `"bar/app/infra#/metrics/inventory?waffleFilter=(expression:'host.ip%20%3A%20151.101.202.217',kind:kuery)"`;

exports[`getMetricsHref getMetricsIpHref does not specify a base path when none is available 1`] = `"/app/infra#/metrics/inventory?waffleFilter=(expression:'host.ip%20%3A%20151.101.202.217',kind:kuery)"`;

exports[`getMetricsHref getMetricsIpHref returns a url for ors between multiple ips 1`] = `"foo/app/infra#/metrics/inventory?waffleFilter=(expression:'host.ip%20%3A%20152.151.23.192%20or%20host.ip%20%3A%20151.101.202.217',kind:kuery)"`;

exports[`getMetricsHref getMetricsKubernetesHref creates a link for valid parameters 1`] = `"foo/app/infra#/link-to/pod-detail/test-pod-uid"`;

exports[`getMetricsHref getMetricsKubernetesHref does not specify a base path when none is available 1`] = `"/app/infra#/link-to/pod-detail/test-pod-uid"`;

exports[`getMetricsHref getMetricsKubernetesHref selects the first pod uid when there are multiple 1`] = `"/app/infra#/link-to/pod-detail/test-pod-uid"`;
